pkg/compute/deploy.go

```diff
@@ -221,3 +226,39 @@ func (c *Client) UpdatePackage(serviceID string, v int, path string) error {
 
 	return nil
 }
+
+// GetLatestIdealVersion gets the most ideal service version using the following logic:
+// - Find the active version and return
+// - If no active version, find the latest locked version and return
+// - Otherwise return the latest version
+//
+// Note: assumes that the provided list is chronoligically sorted, such as that
```

Comment:
I think this implementation is Good Enough™️  for now but relying on chronological ordering (an analog of creation time by way of auto-incrementing version number) may have unintended consequences.

Using `updated_at` for ordering explicitly is probably the more correct behavior—the assumption being that users would want to use the latest version that changed, and not necessarily the latest version that exists. Hard to say.

Reply:
👍 Agreed. Have implemented sorting by `UpdatedAt` in https://github.com/fastly/cli/pull/50/commits/4cfd2cef3a2d97e199f97b663fa5ad04ed56a245